builder/dockerfile/evaluator.go

```diff
@@ -56,10 +58,32 @@ var allowWordExpansion = map[string]bool{
 	command.Expose: true,
 }
 
-var evaluateTable map[string]func(*Builder, []string, map[string]bool, string) error
+type dispatchRequest struct {
+	builder    *Builder // TODO: replace this with a smaller interface
```

Comment:
Hmm, I'm not sure what you mean. I guess right now our "whole build" struct is `parser.Result()` + `options` ?

This object is meant to pass in just the parts that are needed to run the dispatch functions. Does any individual dispatch function need access to the entire build definition?

My comment was more for things like `builder.docker` (the backend). Instead of passing around the massive `Builder` object, we could just pass a struct with a much smaller interface that provides just the 3 or 4 things needed by dispatchers.